public/pages/AnomalyCharts/containers/AnomalyHeatmapChart.tsx

```diff
@@ -109,28 +125,57 @@ export const AnomalyHeatmapChart = React.memo(
       });
 
       return [
-        COMBINED_OPTIONS,
+        getViewableCombinedOptions(
+          COMBINED_OPTIONS,
+          props.heatmapDisplayOption?.entityOption
+        ),
         {
           label: 'Individual entities',
-          options: individualEntityOptions,
+          options: individualEntityOptions.reverse(),
         },
       ];
     };
 
+    const getViewableCombinedOptions = (
+      existingOptions: any,
+      selectedCombinedOption: any | undefined
+    ) => {
+      if (!selectedCombinedOption) {
+        return existingOptions;
+      }
+      return {
+        label: existingOptions.label,
+        options: uniq([selectedCombinedOption, ...existingOptions.options]),
+      };
+    };
+
     const [originalHeatmapData, setOriginalHeatmapData] = useState(
-      getAnomaliesHeatmapData(
-        props.anomalies,
-        props.dateRange,
-        AnomalyHeatmapSortType.SEVERITY,
-        COMBINED_OPTIONS.options[0].value
-      )
+      props.isNotSample
+        ? // use anomaly summary data in case of realtime result
+          getEntitytAnomaliesHeatmapData(
+            props.dateRange,
+            props.entityAnomalySummaries,
+            props.heatmapDisplayOption.entityOption.value
+          )
+        : // use anomalies data in case of sample result
+          getAnomaliesHeatmapData(
+            props.anomalies,
```

Comment:
I am not worried about checking because `anomalies` is set at only 1 [place](https://github.com/opendistro-for-elasticsearch/anomaly-detection-kibana-plugin/blob/master/public/pages/AnomalyCharts/containers/AnomaliesChart.tsx#L86), and it won't be undefined.

After second thought, I think I can make change to check if it exists or not when converting it to heatmap data during `getAnomaliesHeatmapData`, and `props.anomalies` is being only used in `getAnomaliesHeatmapData`